be/src/storage/rowset/segment_v2/page_io.cpp

```diff
@@ -195,6 +198,57 @@ Status PageIO::read_and_decompress_page(const PageReadOptions& opts, PageHandle*
         opts.stats->uncompressed_bytes_read += body_size;
     }
 
+    if (opts.use_page_cache &&
+        (opts.encoding_type == EncodingTypePB::DICT_ENCODING || opts.encoding_type == EncodingTypePB::BIT_SHUFFLE)) {
+        size_t dict_header_size = BINARY_DICT_PAGE_HEADER_SIZE;
```

Comment:
better to move this logic to a function

Reply:
> 

fix in https://github.com/StarRocks/starrocks/pull/1597/commits/a276d0c37f8372f57bb86a715c774895905d9772, PTAL